src/viam/ur/module/ur_arm.cpp

```diff
@@ -640,36 +650,50 @@ void URArm::move_(std::vector<Eigen::VectorXd> waypoints, std::chrono::milliseco
     VIAM_SDK_LOG(info) << "move: get_joint_positions end" << unix_time.count();
 
     VIAM_SDK_LOG(info) << "move: compute_trajectory start " << unix_time.count();
-    const Eigen::VectorXd curr_waypoint_deg =
-        Eigen::VectorXd::Map(curr_joint_pos.data(), boost::numeric_cast<Eigen::Index>(curr_joint_pos.size()));
-    const Eigen::VectorXd curr_waypoint_rad = curr_waypoint_deg * (M_PI / 180.0);
+    auto curr_waypoint_deg = Eigen::VectorXd::Map(curr_joint_pos.data(), boost::numeric_cast<Eigen::Index>(curr_joint_pos.size()));
+    auto curr_waypoint_rad = degrees_to_radians(std::move(curr_waypoint_deg)).eval();
     if (!curr_waypoint_rad.isApprox(waypoints.front(), k_waypoint_equivalancy_epsilon_rad)) {
-        waypoints.insert(waypoints.begin(), curr_waypoint_rad);
+        waypoints.emplace_front(std::move(curr_waypoint_rad));
     }
     if (waypoints.size() == 1) {  // this tells us if we are already at the goal
         VIAM_SDK_LOG(info) << "arm is already at the desired joint positions";
         return;
     }
 
-    // calculate dot products and identify any consecutive segments with dot product == -1
-    std::vector<size_t> segments;
-    segments.push_back(0);
-    for (size_t i = 2; i < waypoints.size(); i++) {
-        Eigen::VectorXd segment_AB = (waypoints[i - 1] - waypoints[i - 2]);
-        Eigen::VectorXd segment_BC = (waypoints[i] - waypoints[i - 1]);
-        segment_AB.normalize();
-        segment_BC.normalize();
-        const double dot = segment_BC.dot(segment_AB);
+    // Walk all interior points of the waypoints list, if any. If the
+    // point of current interest is the cusp of a direction reversal
+    // w.r.t. the points immediately before and after it, then splice
+    // all waypoints up to and including the cusp point into a new
+    // segment, and then begin accumulating a new segment starting at
+    // the cusp point. The cusp point is duplicated, forming both the
+    // end of one segment and the beginning of the next segment. After
+    // exiting the loop, any remaining waypoints form the last (and if
+    // no cusps were identified the only) segment. If one or more cusp
+    // points were identified, the waypoints list will always have at
+    // least two residual waypoints, since the last waypoint is never
+    // examined, and the splice call never removes the waypoint being
+    // visited.
+    //
+    // NOTE: This assumes waypoints have been de-duplicated to avoid
+    // zero-length segments that would cause numerical issues in
+    // normalized() calculations.
+    std::vector<decltype(waypoints)> segments;
```

Comment:
Same here with initialization

Reply:
Default initialization for containers is fine.